src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -148,6 +154,15 @@ int dhcp6_lease_get_iaid(sd_dhcp6_lease *lease, be32_t *iaid) {
         return 0;
 }
 
+int dhcp6_lease_set_duid(sd_dhcp6_lease *lease, struct duid *duid, size_t duid_len) {
+        assert_return(lease, -EINVAL);
+        assert_return(duid, -EINVAL);
+
+        lease->duid = duid;
+        lease->duid_len = duid_len;
```

Comment:
I thought about keeping a copy, but DUID does not change once it is set (either from configuration or from systemd generation) unless networkd restarts, in which case new set of objects are allocated. So it looks safe to use a pointer to a single copy. On second thoughts, I'll change it back to keep a copy to be consistent with how DHCPv4 stores client-id.